src/gestures/use-pan-gesture.ts

```diff
@@ -360,35 +355,55 @@ export interface PanHandlers {
     ): void
 }
 
-type MotionXY = { x: MotionValue<number>; y: MotionValue<number> }
+const noop = () => {}
 
 /**
  *
  * @param handlers -
  * @param ref -
  * @internal
  */
-export function usePanGesture(
-    handlers: PanHandlers,
-    ref: RefObject<Element>
-): undefined
-export function usePanGesture(
-    handlers: PanHandlers
-): { onPointerDown: EventHandler }
 export function usePanGesture(
     { onPan, onPanStart, onPanEnd, onPanSessionStart }: PanHandlers,
-    ref?: RefObject<Element>
+    ref: RefObject<Element>
 ) {
+    const hasPanEvents = onPan || onPanStart || onPanEnd || onPanSessionStart
     const session = useRef<EventSession | null>(null)
-    const pointer = useRef<MotionXY | null>(null)
     const lastMoveEvent = useRef<MouseEvent | TouchEvent | PointerEvent | null>(
         null
     )
     const lastMoveEventInfo = useRef<EventInfo | null>(null)
     const { transformPagePoint } = useContext(MotionPluginContext)
 
-    const getPanInfo = ({ point }: EventInfo): PanInfo => {
+    const pointerEventSubscription = useRef<RemoveEvent | null>(null)
+
+    // Group these as a new hook to enable either an early return or one after making
+    // the pan gestures.
+    function useAttachPanListeners(
+        cancelPanHandler: RemoveEvent,
+        onPointerDownHandler?: EventHandler
+    ) {
+        usePointerEvent(ref, "pointerdown", onPointerDownHandler)
+        useUnmountEffect(cancelPanHandler)
+    }
+
+    // Early return if we have no pan events
+    // Note: No more hooks other than useAttachPanListeners permitted past here.
```

Comment:
This seems to be a bit risky. how do we ensure that this is enforced? Is there a test covering both paths?

Reply:
True but I think it's the only way to early return. I'll revert it because you're right and it's tricky to test the alternative path.

Reply:
Actually thinking about it we do test the early return path because it uncovered a bug where I didn't actually `return` - but I still think we should revert as it's an odd pattern.